contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,61 @@
+# Managing cStor based OpenEBS Volumes. 
+
+## Background 
+
+
+## Design
+
+Provisioning of cStor based Volume involve two steps. 
+
+1. Creating cStor StoragePools:
+
+   * Admin will create SPC that contains information like:
+     - type=block
+     - name
+     - disks list (type of PDC or specific disks (PD) )
+     - resource requests (optional, like CPU, RAM)
+     - raid type (optional, default _stripe_, valid values: _stripe_, _mirror_)
+     - nodes list (optional, default: _all_)
+     - max-pools (optional, default: _none_)
+     
+     Persistent Disk and Persistent Disk Claims are disks and disk claims that are handled by the nodebot, that runs on each of the node and uniquely identifies the disk attached to a node. The uniquness is maintained across the clusters. The way PDs are created and identified will be part of the node-bot design document. It suffices here to say that a PD will be identified by a cluster wide unique identifier like -- rs://host:nodea/device:/dev/disk-by-id/avcd-121312-hfjgdf. (TODO Kiran : Link the nodebot design document here.)
+
+   * maya-cstor-operator (could be embedded into maya-apiserver), will be watching for SPCs (type=block).
+   
+   * When maya-cstor-operator detects a new SPC object, it will identify the list of nodes that satisfy the SPC constraints in terms of:
+     - availability of disks
+     - resources (CPU and RAM) 
+     - nodes filter. 
+     This step can result in more than one node satisfying the constraints. Only the number of nodes required (as specified using max-pools) will be picked up. 
+     
+   * For each of the potentional node where the cStor pool can be created, maya-cstor-operator will:
+     * create a cStorPool (CRD), which will include the following information:
+       - unique id
+       - name
+       - actual disks paths to be used. 
+       - raid type
+     * create a Deployment YAML file that contains the cStor container and its associated sidecars. The cStor-Sidecar is passed the “unique id” of the cStorPool (CRD) 
+     * create SP object
+
+
+2. Creating Volume using the (cStor) Storage Pools:
+
+   * Admin will create a PVC that is associated to StoragePool (linked by the pool name). 
+
+   * maya-apiserver will create the OV Controller and the OV Service (to get the portal ip address). The OV Controller will be passed a ConfigMap (that will be mounted into the OV Controller). The ConfigMap will contain the following details:
```

Comment:
if I'm not wrong, this will be updated as CRD rather than configmap, by having a sidecar for target side also.